### ISSUE TITLE
linchpin validate does not validate when the credential section is under-indented 

### ISSUE THREAD

**Issue by samvarankashyap** [CentOS-PaaS-SIG/linchpin#774]
$subject 
example : 

```
---
topology_name: openshift
resource_groups:
  - resource_group_name: test1
    resource_group_type: openshift
    resource_definitions:
      - name: openshift
        role: openshift_inline
        data:
          - apiVersion: v1
            kind: ReplicationController
            metadata:
              name: jenkins-slave
              namespace: central-ci-test-ghelling
            spec:
              replicas: 7
              selector:
                name: jenkins-slave
              template:
                metadata:
                  labels:
                    name: jenkins-slave
                spec:
                  containers:
                    - image: redhatqecinch/jenkins_slave:latest
                      name: jenkins-slave
                      env:
                        - name: JENKINS_MASTER_URL
                          value: http://10.8.172.6/
                        - name: JSLAVE_NAME
                          value: mynode
                  restartPolicy: Always
                  securityPolicy:
                    runAsUser: 1000090000
  credentials:
    filename: name_of_credsfile.yaml  # fetched from --creds-path is provided
    profile: name_of_profile # defaults to 'default' profile in cred_file
    # alternative way when using --cred-path
    #filename: openshift.yaml
    #profile: test
```

**Comment by 14rcole** [comment]
Our schema validation may need to be reworked in order to support this, I think it's too myopic right now.  We only validate within each `resource_definition` individually

**Comment by 14rcole** [comment]
Fixed by #892
